content/blog/writing-better-angular-tests/index.md

```diff
@@ -0,0 +1,208 @@
+
+## Transitioning to Jest and Angular Testing Library
+
+### Get rid of Karma
+
+Angular ships with Karma alongside Jasmine for running tests and collecting coverage. With Jest an Angular project no longer need Karma or the other packages that would be installed by the Angular CLI.
+
+#### Uninstall Karma
+
+```bash
+npm uninstall karma karma-chrome-launcher karma-coverage-istanbul-reporter karma-jasmine karma-jasmine-html-reporter
+```
+
+#### Remove the leftover configurations
+
+Deleting the following will remove the leftover configuration files from the project:
+
+```bash
+karma.config.js
+src/test.ts
+```
+
+Once those two files are deleted any references to `src/test.ts` will need to be removed. Removing the paths from the following file that reference them cleans it up easily enough:
+
+```json
+tsconfig.spec.json
+{
+  ...,
+  "files": [
+    "src/test.ts", <- delete
+    ...
+    ]
+}
+```
+
+The project also no longer needs the `test` key inside of `angular.json` and thus it can be removed.
+
+```json
+angular.json
+{
+  ...,
+  "test": {...} <- delete
+  ....
+}
+```
+
+Finally the project no longer needs the Jasmine types in the spec configuration
+
+```json
+tsconfig.spec.json
+{
+  ...,
+  "compilerOptions": {
+    ...,
+    "types": [
+      "jasmine", <- delete
+      ...
+    ]
+  }
+}
+```
+
+Now the project is ready for installing any other test runner.
+
+### Setting up Jest
+
+Now that the project has no Karma it can be setup with Jest
+
+#### Install Jest
+
+```bash
+npm i -D @types/jest jest jest-preset-angular ts-jest
+```
+
+This installs Jest, the types for Jest, a TypeScript pre-processor for Jest, and a preset that makes setting up Jest much easier.
+
+#### Configure Jest
+
+The project now needs to know how to best utilize Jest. Creating and modify the following files will allow Jest to load it's own configuration.
+
+```js
+jest.config.js
+module.exports = {
+    preset: 'jest-preset-angular',
+    setupFilesAfterEnv: [
+        '<rootDir>/jest.setup.ts'
+    ]
+};
+```
+
+```typescript
+jest.setup.ts
+import 'jest-preset-angular';
+```
+
+```json
+tsconfig.spec.json
+{
+  ...,
+  "compilerOptions": {
+    ...,
+    "types": [
+      "jest", <- new
+      ...
+    ]
+  }
+}
+```
+
+```json
+tsconfig.json
+{
+  ...,
+  "compilerOptions": {
+    ...,
+    "esModuleInterop": true, <- new
+    "emitDecoratorMetadata": true, <- new
+    ...
+  },
+  ...
+}
+```
+
+```json
+package.json
+{
+  ...,
+  "scripts": {
+    ...,
+    "test": "jest --coverage --config ./jest.config.js", <- new
+    "test:watch": "jest -o --watch --config ./jest.config.js", <- new
+    ...
+  },
+  ...
+}
+```
+
+Jest is now the test runner for the project. It can now be used to run the tests, but even better it can be used in combination with Testing Library.

```

Comment:
```suggestion
Jest is now the test runner for the project. It can now be used to run the tests, but even better, it can be used in combination with Testing Library.
```